modules/Kconfig.tfm

```diff
@@ -6,17 +6,49 @@
 
 
 config TFM_BOARD
-	string
-	default "nordic_nrf/nrf9160dk_nrf9160" if BOARD_NRF9160DK_NRF9160NS
-	default "nordic_nrf/nrf5340dk_nrf5340_cpuapp" if BOARD_NRF5340DK_NRF5340_CPUAPPNS
-	default "nordic_nrf/nrf5340pdk_nrf5340_cpuapp" if BOARD_NRF5340PDK_NRF5340_CPUAPPNS
-	default "nxp/lpcxpresso55s69" if BOARD_LPCXPRESSO55S69_CPU0
-	default "mps2/an521" if BOARD_MPS2_AN521
-	default "stm/nucleo_l552ze_q" if BOARD_NUCLEO_L552ZE_Q
-	default "musca_b1" if BOARD_MUSCA_B1
+	bool
 	help
-	  The board name used for building TFM. Building with TFM requires that
-	  TFM has been ported to the given board/SoC.
+	  Hidden option signifying that TF-M has been ported to
+	  the corresponding platform and therefore it is supported
+	  on the given Zephyr board.
+
+# List of hidden BOARD options corresponding to Zephyr
+# boards that are supported in the TF-M project.
+
+config TFM_BOARD_LPCXPRESSO55S69
+	bool
+	depends on BOARD_LPCXPRESSO55S69_CPU0 && TRUSTED_EXECUTION_NONSECURE
```

Comment:
Is there actually a need to have these dependencies also here in `TFM_BOARD_*` options? As @tejlmand points out in https://github.com/zephyrproject-rtos/zephyr/pull/30145#issuecomment-730962083, more than one Zephyr board could potentially use the same TF-M platform.
And I agree with @tejlmand that `TFM_PLATFORM*` seems to be a more appropriate naming.

Reply:
Can we avoid duplicating `depends on TRUST_EXECUTION_NONSECURE` by moving it to TFM_PLATFORM?